src/parsing.rs

```diff
@@ -1,198 +1,289 @@
-  let media_type = params.media_type;
-  let syntax = params
-    .maybe_syntax
-    .unwrap_or_else(|| get_syntax(media_type));
-  let (comments, program, tokens) =
-    parse_string_input(input, syntax, params.capture_tokens, parse_mode)
-      .map_err(|err| Diagnostic {
-        display_position: source.line_and_column_display(err.span().lo),
-        specifier: specifier.clone(),
-        message: err.into_kind().msg().to_string(),
-      })?;
-  let program = post_process(program);
-
-  Ok(ParsedSource::new(
-    specifier,
-    params.media_type.to_owned(),
-    source,
-    MultiThreadedComments::from_single_threaded(comments),
-    Arc::new(program),
-    tokens.map(Arc::new),
-  ))
-}
-
-fn parse_string_input(
-  input: StringInput,
-  syntax: Syntax,
-  capture_tokens: bool,
-  parse_mode: ParseMode,
-) -> Result<
-  (SingleThreadedComments, Program, Option<Vec<TokenAndSpan>>),
-  swc_ecmascript::parser::error::Error,
-> {
-  let comments = SingleThreadedComments::default();
-  let lexer = Lexer::new(syntax, TARGET, input, Some(&comments));
-
-  if capture_tokens {
-    let lexer = swc_ecmascript::parser::Capturing::new(lexer);
-    let mut parser = swc_ecmascript::parser::Parser::new_from(lexer);
-    let program = match parse_mode {
-      ParseMode::Program => parser.parse_program()?,
-      ParseMode::Module => Program::Module(parser.parse_module()?),
-      ParseMode::Script => Program::Script(parser.parse_script()?),
-    };
-    let tokens = parser.input().take();
-
-    Ok((comments, program, Some(tokens)))
-  } else {
-    let mut parser = swc_ecmascript::parser::Parser::new_from(lexer);
-    let program = match parse_mode {
-      ParseMode::Program => parser.parse_program()?,
-      ParseMode::Module => Program::Module(parser.parse_module()?),
-      ParseMode::Script => Program::Script(parser.parse_script()?),
-    };
-
-    Ok((comments, program, None))
-  }
-}
-
-/// Gets the default `Syntax` used by `deno_ast` for the provided media type.
-pub fn get_syntax(media_type: MediaType) -> Syntax {
-  match media_type {
-    MediaType::JavaScript => Syntax::Es(get_es_config(false)),
-    MediaType::Jsx => Syntax::Es(get_es_config(true)),
-    MediaType::TypeScript => Syntax::Typescript(get_ts_config(false, false)),
-    MediaType::Dts => Syntax::Typescript(get_ts_config(false, true)),
-    MediaType::Tsx => Syntax::Typescript(get_ts_config(true, false)),
-    _ => Syntax::Es(get_es_config(false)),
-  }
-}
-
-/// Gets the default `EsConfig` used by `deno_ast` for the provided options.
-pub fn get_es_config(jsx: bool) -> EsConfig {
-  EsConfig {
-    class_private_methods: true,
-    class_private_props: true,
-    class_props: true,
-    dynamic_import: true,
-    export_default_from: true,
-    export_namespace_from: true,
-    import_meta: true,
-    jsx,
-    nullish_coalescing: true,
-    num_sep: true,
-    optional_chaining: true,
-    top_level_await: true,
-    ..EsConfig::default()

```

Comment:
I changed these to not use the `default()` anymore. That way nothing will sneak past us (such as `static_blocks: false,` previously being the default here until this PR).